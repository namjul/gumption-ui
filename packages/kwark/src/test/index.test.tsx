import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import { kwark, createHook } from '..';

afterEach(cleanup);

describe('kwark', () => {
  test('renders', () => {
    const Kwark = kwark('div');
    const { asFragment } = render(<Kwark>Hello World</Kwark>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('render `style` attribute', () => {
    const Kwark = kwark('div');
    const { asFragment } = render(
      <Kwark style={{ color: 'red' }} className="hello-world">
        Hello World
      </Kwark>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('render wrapElement using `useHook` prop', () => {
    const Anchor = kwark('a');
    type AnchorHTMLProps = React.HTMLAttributes<any> &
      React.AnchorHTMLAttributes<any> & {
        /**
         * Function returned by the hook to wrap the element to which html props
         * will be passed.
         */
        wrapElement?: (element: React.ReactNode) => React.ReactNode;
      };
    const useHook = createHook<{ myProp: string }, AnchorHTMLProps>({
      name: 'Kwark',
      keys: ['myProp'],
      useProps: ({ myProp }, htmlProps) => ({
        ...htmlProps,
        wrapElement: (element) => <Anchor>:{element}:</Anchor>,
        className: `${htmlProps.className} ${myProp}`,
      }),
    });
    const Kwark = kwark('div', {
      useHook,
    });
    const { asFragment } = render(
      <Kwark className="child-element" myProp="world">
        Hello
      </Kwark>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
