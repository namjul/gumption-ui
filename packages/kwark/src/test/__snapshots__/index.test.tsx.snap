// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kwark render \`style\` attribute 1`] = `
<DocumentFragment>
  <div
    style="color: red;"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`kwark render with \`css\` prop 1`] = `
<DocumentFragment>
  <div
    class="_hwfcu5"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`kwark render with \`useHook\` prop 1`] = `
<DocumentFragment>
  <a>
    :
    <div
      class="_hwfcu5"
    >
      Hello
    </div>
    :
  </a>
</DocumentFragment>
`;

exports[`kwark render with merged className 1`] = `
<DocumentFragment>
  <div
    class="satori _hwfcu5"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`kwark renders 1`] = `
<DocumentFragment>
  <div>
    Hello
  </div>
</DocumentFragment>
`;
